fecgov/openFEC
# [WIP] fix flake8 and prettier errors

## Summary (required)

- Resolves #5493 

This ticket corrects all flake8, prettier, and eof styling errors. I also added pre-commit check to our circleci pipeline. 

### Required reviewers 2-3 developers 

## Impacted areas of the application

General components of the application that this PR will affect:

-  circleci pipeline 

## How to test

Test pre-commit 
1. checkout this branch 
2. activate virtualenv 
3. ` pre-commit --version`
4. If you get error: `pre-commit install `
5. `pre-commit run --all-files` ensure that all tests pass

Test logic changes in webservices/resources/candidates.py and webservices/resources/committees.py. I switched to using [.is_](https://docs.sqlalchemy.org/en/20/core/operators.html) to remove flake8 error.

6. `flask run`
7. Test committees and candidates endpoints and compare to prod. 

Sample URLs: 

http://127.0.0.1:5000/v1/committees/?year=2024
https://api.open.fec.gov/v1/committees/?year=2024&api_key=DEMO_KEY

http://127.0.0.1:5000/v1/candidates/?is_active_candidate=true
https://api.open.fec.gov/v1/candidates/?is_active_candidate=true&api_key=DEMO_KEY

Test pre-commit in circleci 

8. create a new branch based off of this one 
9. Make any formatting error 
10. commit `git commit -a --no-verify -m "message"` 
11.  push and ensure that circleci build fails.
